src/deploy/osp_deployer/settings/gen_scripts/gen_ini_file.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python
+
+# Copyright (c) 2018 Dell Inc. or its subsidiaries.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# Run this script run from the director node as the director's admin user.
+# This script assumes the update_ssh_config.py is present.
+###############################################################################
+
+# IMPORTS
+from collections import defaultdict
+import sys
+
+
+def read_input_file(input_file):
+    idata = {}
+    with open(input_file, 'r') as f:
+        next(f)  # skip first line
+        for line in f:
+            line = line.strip()
+            if len(line) > 2:
+                fields = line.split("|")
+                if len(fields) == 2:
+                    idata[fields[0]] = fields[1]
+    return idata
+
+
+def generate_output_file(idata, input_ini_file, output_ini_file):
+    with open(input_ini_file, 'r') as f, open(output_ini_file, 'w') as fnew:
+        for line in f:
+            data = line.strip()
+            if data.startswith("#") or data == "":
+                fnew.write(line)
+            else:
+                if "=" in data:
+                    okey, oval = data.split('=', 1)
```

Comment:
Does this code still work if equal sign is first character or last character?

Reply:
Yes, it has no harmful affects.